requests/cookies.py

```diff
@@ -292,9 +292,11 @@ def multiple_domains(self):
         """
         domains = []
         for cookie in iter(self):
-            if cookie.domain is not None and cookie.domain in domains:
-                return True
             domains.append(cookie.domain)
+            domains_set = set(domains)
```

Comment:
A few quick notes here.

We’re rebuilding the set every loop which probably isn’t necessary. Since it’s serving the same purpose of `domains`, what do you think about making `domains` a set?

The other issue I’m seeing is the conditional. Let’s take an example set of domains `['google.com', None, 'google.com']`. The first two loops will add both 'google.com' and None to `domains` and the third loop will satisfy the criteria despite there only being one domain. It may be better if we only add if the cookie.domain is not None and then have the single check for `len(domains)` being > 1

Reply:
I think you're correct there, I'll update the pr tomorrow.